src/main/java/org/altbeacon/beacon/Region.java

```diff
@@ -161,19 +160,27 @@ public boolean matchesBeacon(Beacon beacon) {
         return true;
 	}
 
-	@Override
-	public int hashCode() {
-		return this.mUniqueId.hashCode();
-	}
+    @Override
+    public int hashCode() {
+        int result = mIdentifiers.hashCode();
+        result = 31 * result + mUniqueId.hashCode();
+        return result;
+    }
 
-	public boolean equals(Object other) {
-		 if (other instanceof Region) {
-			return ((Region)other).mUniqueId.equals(this.mUniqueId);
-		 }
-		 return false;
-	}
-	
-	public String toString() {
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+
+        Region region = (Region) o;
+
+        if (!mIdentifiers.equals(region.mIdentifiers)) return false;
```

Comment:
I think this line is a problem.  The use of the uniqueId string to identify the region is intended to let you stop ranging or monitoring the region (or update the region being monitored/ranged) by passing in a second region with the same uniqueId string.  iOS CoreLocation uses the same basic mechanism.  If you construct a new region object with a different identifier set and want the APIs to range/monitor the modified region instead, it is supposed to *replace* that region in the ranging/monitoring list.  This wouldn't work anymore with this line added.